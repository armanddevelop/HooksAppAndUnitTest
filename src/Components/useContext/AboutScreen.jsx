import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const userRest = {};
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
      <button className="btn btn-warning" onClick={() => setUser(userRest)}>
        Log out
      </button>
    </div>
  );
};
