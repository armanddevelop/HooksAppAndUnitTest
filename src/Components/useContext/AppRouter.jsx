import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { LognScreen, AboutScreen, HomeScreen, NavBar } from "./index";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route exact path="/" element={<HomeScreen />} />
            <Route exact path="/about" element={<AboutScreen />} />
            <Route exact path="/login" element={<LognScreen />} />
            <Route path="*" element={<HomeScreen />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
};
