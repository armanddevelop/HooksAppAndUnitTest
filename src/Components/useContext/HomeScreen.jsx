import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  //console.log("esto vale userContext ", user);
  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
    </div>
  );
};
