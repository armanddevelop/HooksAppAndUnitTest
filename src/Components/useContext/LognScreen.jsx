import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LognScreen = () => {
  const { setUser } = useContext(UserContext);
  const userObj = {
    id: 123,
    name: "creting",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(userObj)}>
        User
      </button>
    </div>
  );
};
