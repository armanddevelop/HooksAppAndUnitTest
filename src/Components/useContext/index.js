export { HomeScreen } from "./HomeScreen";
export { AboutScreen } from "./AboutScreen";
export { LognScreen } from "./LognScreen";
export { NavBar } from "./NavBar";
