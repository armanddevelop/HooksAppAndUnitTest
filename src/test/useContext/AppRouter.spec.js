import { mount } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { AppRouter } from "../../Components/useContext/AppRouter";
import { UserContext } from "../../Components/useContext/UserContext";

describe("Test to AppRouter component", () => {
  const user = {
    name: "alicha",
    email: "euche@test.com",
  };
  const container = mount(
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  );
  it("Snapshot to AppRouter component", () => {
    expect(toJson(container)).toMatchSnapshot();
  });
});
