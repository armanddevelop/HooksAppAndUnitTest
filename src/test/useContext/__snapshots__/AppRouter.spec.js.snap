// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to AppRouter component Snapshot to AppRouter component 1`] = `
<AppRouter>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <NavBar>
          <nav
            className="navbar navbar-expand-sm navbar-dark bg-dark"
          >
            <div
              className="container-fluid"
            >
              <div
                className="collapse navbar-collapse"
                id="navbarNavAltMarkup"
              >
                <div
                  className="navbar-nav"
                >
                  <NavLink
                    aria-current="page"
                    className="nav-link"
                    exact="true"
                    to="/"
                  >
                    <Link
                      aria-current="page"
                      className="nav-link active"
                      exact="true"
                      to="/"
                    >
                      <a
                        aria-current="page"
                        className="nav-link active"
                        exact="true"
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </Link>
                  </NavLink>
                  <NavLink
                    className="nav-link"
                    exact="true"
                    to="/login"
                  >
                    <Link
                      className="nav-link"
                      exact="true"
                      to="/login"
                    >
                      <a
                        className="nav-link"
                        exact="true"
                        href="/login"
                        onClick={[Function]}
                      >
                        Login
                      </a>
                    </Link>
                  </NavLink>
                  <NavLink
                    className="nav-link"
                    exact="true"
                    to="/about"
                  >
                    <Link
                      className="nav-link"
                      exact="true"
                      to="/about"
                    >
                      <a
                        className="nav-link"
                        exact="true"
                        href="/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                    </Link>
                  </NavLink>
                </div>
              </div>
            </div>
          </nav>
        </NavBar>
        <div
          className="container"
        >
          <Routes>
            <HomeScreen>
              <div>
                <h1>
                  HomeScreen
                </h1>
                <hr />
                <pre />
              </div>
            </HomeScreen>
          </Routes>
        </div>
      </div>
    </Router>
  </BrowserRouter>
</AppRouter>
`;
